e2e/cypress/integration/messaging/permalink_loading_indicator_spec.js

```diff
@@ -0,0 +1,89 @@
+        const privateChannelName = 'test-private-channel-' + dateNow;
+        const publicChannelName = 'test-public-channel-' + dateNow;
+
+        // # Get current team id
+        cy.getCurrentTeamId().then((teamId) => {
+            // # Create private channel to post message
+            cy.apiCreateChannel(teamId, privateChannelName, privateChannelName, 'P', 'Test private channel').then((response) => {
+                const testPrivateChannel = response.body;
+
+                // # Click on test private channel
+                cy.get('#sidebarItem_' + testPrivateChannel.name).click({force: true});
+
+                // # Post message to use
+                cy.postMessage(message);
+
+                // # Get last post id from that postlist area
+                cy.getLastPostId().then((postId) => {
+                    // # Click on ... button of last post
+                    cy.clickPostDotMenu(postId);
+
+                    // # Click on permalink option
+                    cy.get(`#permalink_${postId}`).click();
+
+                    // # Copy permalink from button
+                    cy.get('#linkModalCopyLink').click();
+
+                    // # Copy link url from text area
+                    cy.get('#linkModalTextArea').invoke('val').as('linkText');
+
+                    // # Close modal dialog
+                    cy.get('#linkModalCloseButton').click();
+                });
+            });
+        });
+
+        // # Get current team id
+        cy.getCurrentTeamId().then((teamId) => {
+            // # Create public channel to post permalink
+            cy.apiCreateChannel(teamId, publicChannelName, publicChannelName, 'O', 'Test public channel').then((response) => {
+                const testPublicChannel = response.body;
+
+                // # Click on test public channel
+                cy.get('#sidebarItem_' + testPublicChannel.name).click({force: true});
+
+                // # Paste link on postlist area
+                cy.get('@linkText').then((linkText) => {
+                    cy.postMessage(linkText);
+
+                    // # Get last post id from that postlist area
+                    cy.getLastPostId().then((postId) => {
+                        // # Click on permalink
+                        cy.get(`#postMessageText_${postId} > p > .markdown__link`).scrollIntoView().click();
+
+                        // * Check if url include the permalink
+                        cy.url().should('include', linkText);
+                    });
+
+                    // # Get last post id from open channel
+                    cy.getLastPostId().then((clickedpostid) => {
+                        // * Check the sent message
+                        cy.get(`#postMessageText_${clickedpostid}`).should('be.visible').and('have.text', message);
+
+                        // * Check the loading spinner does not appear
+                        cy.get('#loadingSpinner').should('not.be.visible');
```

Comment:
It's not the `#loadingSpinner` ('cause this is used when switching into a channel) but the loading indicator that shows up when scrolling above to load more messages.  You may add new ID [here](https://github.com/jgbaylon/mattermost-webapp/blob/mm-18701/components/post_view/post_list_row/post_list_row.jsx#L71).

Not mentioned in the ticket but while here, it would be nice to have the following as well: 
- check whether `Load more messages` is not visible (please add new element ID [here](https://github.com/jgbaylon/mattermost-webapp/blob/mm-18701/components/post_view/post_list_row/post_list_row.jsx#L57))
- check if `channelIntro` is visible and matches the [intro title](https://github.com/jgbaylon/mattermost-webapp/blob/mm-18701/components/post_view/channel_intro_message/channel_intro_message.jsx#L328) (to make sure that we're redirected into the expected channel)

Reply:
Updated in latest commit.